app/viewmodels/regulatorsAndDocuments/GoverningDocumentStatusHelper.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels.regulatorsAndDocuments
+
+import models.UserAnswers
+import models.regulators.SelectGoverningDocument
+import models.regulators.SelectGoverningDocument.Other
+import pages.QuestionPage
+import pages.regulatorsAndDocuments._
+import viewmodels.StatusHelper
+
+class GoverningDocumentStatusHelper extends StatusHelper {
+
+
+  private val allPages: Seq[QuestionPage[_]] =  Seq(SelectGoverningDocumentPage, GoverningDocumentNamePage, IsApprovedGoverningDocumentPage,
+    WhenGoverningDocumentApprovedPage, HasCharityChangedPartsOfGoverningDocumentPage, SectionsChangedGoverningDocumentPage)
+
+  private val validJourneys: Map[String, Seq[QuestionPage[_]]] = Map(
+    ("1", Seq(SelectGoverningDocumentPage, GoverningDocumentNamePage, IsApprovedGoverningDocumentPage, WhenGoverningDocumentApprovedPage)),
+    ("2", Seq(SelectGoverningDocumentPage, GoverningDocumentNamePage, IsApprovedGoverningDocumentPage,
+      WhenGoverningDocumentApprovedPage, HasCharityChangedPartsOfGoverningDocumentPage)),
+    ("3", Seq(SelectGoverningDocumentPage, GoverningDocumentNamePage, IsApprovedGoverningDocumentPage,
+      WhenGoverningDocumentApprovedPage, HasCharityChangedPartsOfGoverningDocumentPage, SectionsChangedGoverningDocumentPage)),
+    ("1A", Seq(SelectGoverningDocumentPage, IsApprovedGoverningDocumentPage, WhenGoverningDocumentApprovedPage)),
+    ("2A", Seq(SelectGoverningDocumentPage, IsApprovedGoverningDocumentPage, WhenGoverningDocumentApprovedPage,
+      HasCharityChangedPartsOfGoverningDocumentPage)),
+    ("3A", Seq(SelectGoverningDocumentPage, IsApprovedGoverningDocumentPage, WhenGoverningDocumentApprovedPage,
+      HasCharityChangedPartsOfGoverningDocumentPage, SectionsChangedGoverningDocumentPage))
+  )
+
+  override def checkComplete(userAnswers: UserAnswers): Boolean = {
+
+    def pageList(documentType:SelectGoverningDocument, list: Seq[QuestionPage[_]]): Seq[QuestionPage[_]] = {
+      if(documentType==Other) list ++ Seq(GoverningDocumentNamePage) else list
+    }
+
+    def notAdditionalPagesDefined(documentType:SelectGoverningDocument, journey:String): Boolean ={
+      val number = if(documentType==Other) journey else s"${journey}A"
+      !(validJourneys.get(number) match {
+        case Some(x) =>
+          userAnswers.oneOfPageIsDefined(allPages.diff(x))
+        case _ => false
```

Comment:
here I feel like it would make it more readable if we were to not apply the ! operand on this match, but instead do
```
case Some(pagesNeededForJourney) => userAnswers.allPagesAreEmpty(allPages.diff(pagesNeededForJourney))
case _ => true
```

where allPagesAreEmpty is pages.forall(page => checkDataPresent(page).isEmpty)